### ISSUE TITLE
Putting "*" and  "!//" in .ignore file would crash Atom

### ISSUE THREAD

**Issue by IceTea233** [atom/atom#25410]
### Prerequisites

* [ ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: <https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode>
    * Followed all applicable steps in the debugging guide: <https://flight-manual.atom.io/hacking-atom/sections/debugging/>
    * Checked the FAQs on the message board for common solutions: <https://discuss.atom.io/c/faq>
    * Checked that your issue isn't already filed: <https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom>
    * Checked that there is not already an Atom package that provides the described functionality: <https://atom.io/packages>

### Description

If some opened project contains a .ignore file and somehow `*` and `!//` were accidently added to the .ignore file at the same time, Atom would crash and cannot open again once the .ignore file is not fixed manually.

### Steps to Reproduce

1. Open a project and create a .ignore file for the project.
2. Add `*` and `!//` to the .ignore file, then save.
3. Atom would crash immediately.

![Atom_crash-1](https://user-images.githubusercontent.com/47026460/163407428-d1ae430f-37e7-4b45-9ff5-1135e3b86108.png)
![Atom_crash-1](https://user-images.githubusercontent.com/47026460/163407553-1d8e903d-b453-46c4-92ce-b56a5dd3fa2c.png)

**Expected behavior:**

(Empty)

**Actual behavior:**

After Atom crashed because of the operations above, it cannot be activated successfully until the .ignore file is revised. 

**Reproduces how often:**

Always.

### Versions

```
Atom    : 1.60.0
Electron: 9.4.4
Chrome  : 83.0.4103.122
Node    : 12.14.1
```
### Additional Information

(Empty)

**Comment by darangi** [comment]
Have you tried reproducing this in safe mode? https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode

**Comment by IceTea233** [comment]
Yes, I have reproduced in safe mode both in "Windows 10 21H2" and "Ubuntu 20.04.4 LTS".

They all lead to the same result.

> Have you tried reproducing this in safe mode? https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
